preprocessing.py

```diff
@@ -0,0 +1,164 @@
+# Authors : Abel Salmona
+
+# those are the function I applied to the data folders before lauching the nnUNet training
+
+# flake8: noqa
+
+import os
+from pathlib import Path
+import sys, argparse, textwrap
+import multiprocessing as mp
+from functools import partial
+from tqdm.contrib.concurrent import process_map
+import nibabel as nib
+import numpy as np
+import torchio as tio
+from image import Image
+import time
+
+# reorientation of the images
+
+def reorient(path):
+    # Load the image using the custom Image class
+    img = Image(param=path)
+
+    # Specify the desired orientation (for example, 'RPI' for Right-Posterior-Inferior)
+    new_orientation = "LAS"
+
+    # Change the orientation of the image
+    img.change_orientation(new_orientation)
+
+    # Save the oriented image
+    img.save(path)
+
+    print(f"Image orientation changed and saved to {path}")
+
+
+# apply to a directory
+def apply_reorient_to_files(directory):
+    t1 = time.time()
+    # visits all files in the given directory
+    for root, dirs, files in os.walk(directory):
+        for file in files:
+            file_path = os.path.join(root, file)
+            if file_path.endswith('.nii.gz'):
+                reorient(file_path)
+            # calls change_orientation on each file   
+    t2 = time.time()
+    print(f"Reorientation of {directory} done in {t2-t1} seconds")  
+
+# here I applied it to imagesTs, labelsTr and imagesTr
+apply_reorient_to_files("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/imagesTs")
+apply_reorient_to_files("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/labelsTr")
+apply_reorient_to_files("C:/Users/abels/OneDrive/Documents/NeuroPoly/canal_seg/segmentation/training/data/datasets/Dataset011_clean/imagesTr")
```

Comment:
the hardcoded paths are not ideal. try to think how to replace them, e.g., using input args.